strax/storage/rucio.py

```diff
@@ -0,0 +1,41 @@
+import json
+import hashlib
+import os.path as osp
+
+import strax
+from strax.storage.files import dirname_to_prefix
+
+export, __all__ = strax.exporter()
+
+
+@export
+class rucio(strax.StorageBackend):
+    """Get data from a rucio directory
+    """
+
+    def get_metadata(self, dirname, **kwargs):
+        dirname = str(dirname)
+        prefix = dirname_to_prefix(dirname)
+        metadata_json = f'{prefix}-metadata.json'
+        fn = rucio_path(metadata_json, dirname)
+        with open(fn, mode='r') as f:
+            return json.loads(f.read())
+
+    def _read_chunk(self, dirname, chunk_info, dtype, compressor):
+        #print('yes')
+        fn = rucio_path(chunk_info['filename'], dirname)
+        return strax.load_file(fn, dtype=dtype, compressor=compressor)
+
+    def _saver(self, dirname, metadata):
+        raise NotImplementedError(
+            "Cannot save directly into rucio, upload with admix instead")
+
+
+def rucio_path(filename, dirname):
```

Comment:
This bit is a bit hard to follow without comments (sorry I'm not an expert in the rucio naming convention). What do the paths look like and are we sure its always these hard-coded conversions?

Reply:
Te path look like ('/dali/lgrandi/rucio/xnt_008710/1f/fb/peaklets-b7dgmtzaef-000047'), It is a hard-coded convention on the Rucio code, you can see it here https://github.com/rucio/rucio/blob/671fe6253981eb632aae3c9ddfe54eb83e63fd1a/lib/rucio/rse/protocols/protocol.py#L114